frontend/src/lib/components/LemonCalendar/LemonCalendar.tsx

```diff
@@ -0,0 +1,141 @@
+import './LemonCalendar.scss'
+import React, { useEffect, useState } from 'react'
+import { dayjs } from 'lib/dayjs'
+import { range } from 'lib/utils'
+import { LemonButton, LemonButtonProps } from 'lib/components/LemonButton'
+import { IconChevronLeft, IconChevronRight } from 'lib/components/icons'
+import clsx from 'clsx'
+
+export interface LemonCalendarProps {
+    /** Fired if a calendar cell is clicked */
+    onClick?: (date: string) => void
+    /** YYYY-MM(-DD) for month that is shown, derived from "value" if absent */
+    firstMonth?: string
+    /** Called if the user changed the month in the calendar */
+    onFirstMonthChanged?: (date: string) => void
+    /** Return the classnames for a date */
+    getLemonButtonProps?: (date: string, month: string, defaultProps: LemonButtonProps) => LemonButtonProps
+    /** Number of months */
+    months?: number
+    /** Number of weeks in each month */
+    weeks?: number
+}
+
+const dayLabels = ['su', 'mo', 'tu', 'we', 'th', 'fr', 'sa']
+
+export function LemonCalendar(props: LemonCalendarProps): JSX.Element {
```

Comment:
should we have some cypress tests on this?

tricky end of year but mid week roll over 
weeks with Feb 29th

Reply:
I'm not sure that's needed. All the complex date wrangling is delegate to dayjs. I basically just iterate and +1 what it gives me. I did add a bunch of `react testing library` tests to cover basic interactions though.